_posts/2021/2021-03-31-hmat_memkind.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Memkind support for heterogeneous memory attributes
+author: michalbiesek
+layout: post
+identifier: hmat_memkind
+---
+
+### Introduction
+
+Memkind is a library that provides support for Persistent Memory. However, it is not the only type of memory that memkind supports.
+It is a general solution designated for the platforms with heterogeneous memory.
+
+But before we go to heterogenous memory itself, let's start with a short recap about [NUMA][kernel-numa].
+The NUMA concept aims the problem of the dynamic extension of CPU count per socket and system memory.
+Before NUMA, commonly used model was uniform memory access (UMA), where all processors share the physical memory uniformly.
+With the UMA approach and rapid growth of number of processors and memory in multi-socket machines, platforms now face scalability problems.
+Single shared bus between all CPU(s) and DRAM memory become a bottleneck, as each CPU decreases the available bandwidth.
+To handle each new device, the length of the bus must be extended, what in turn impacts the latency.
+
+To overcome the challenges mentioned above, engineers decide to introduce a new strategy: None Unified Memory Access(NUMA).
```

Comment:
what engineers? (I think it would be best to just use passive voice and say "a new strategy was introduce: ...)

Non Uniform